src/OpenColorIO/Config.cpp

```diff
@@ -1137,23 +1147,73 @@ ConstConfigRcPtr Config::CreateFromFile(const char * filename)
         return CreateFromBuiltinConfig(match.str(1).c_str());
     }
 
-    std::ifstream istream = Platform::CreateInputFileStream(filename, std::ios_base::in);
-    if (istream.fail())
+    std::ifstream ifstream = Platform::CreateInputFileStream(
+        filename, 
+        std::ios_base::in | std::ios_base::binary
+    );
+
+    if (ifstream.fail())
     {
         std::ostringstream os;
         os << "Error could not read '" << filename;
         os << "' OCIO profile.";
         throw Exception (os.str().c_str());
     }
 
-    return Config::Impl::Read(istream, filename);
+    char magicNumber[2] = { 0 };
+    if (ifstream.read(magicNumber, 2))
+    {
+        // Check if it is an OCIOZ archive.
+        if (magicNumber[0] == 'P' && magicNumber[1] == 'K')
```

Comment:
My interpretation of the spec is that the values must always be in that order.
https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT

Reply:
> All values MUST be stored in little-endian byte order unless otherwise specified in this document for a specific data element.

Yeah looks like it! Thanks for the link.